lib/wallet/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -775,6 +775,7 @@ prop_signTransaction_addsExtraKeyWitnesses
             sealedTx' = signTransaction tl
                 (AnyCardanoEra era)
                 (lookupFnFromKeys extraKeys)
+                Nothing
```

Comment:
The presence of these `Nothing`s seems like we might be introducing a case of algebraic blindness:

https://github.com/quchen/articles/blob/master/algebraic-blindness.md

Can this be replaced with a data type, so instead of writing `Nothing`, we can write something more explicit at call sites?

Reply:
Please let's defer this to https://input-output.atlassian.net/browse/ADP-2675